docs/source/getting-started/app-architecture.rst

```diff
@@ -59,14 +59,17 @@ The last part of the DAML model is the operation to add friends, called a *choic
   :end-before: -- ADDFRIEND_END
 
 DAML contracts are *immutable* (can not be changed in place), so the only way to "update" one is to archive it and create a new instance.
-That is what the ``AddFriend`` choice does: after checking some preconditions, it archives the current user contract and creates a new one with the extra friend added to the list.
+That is what the ``AddFriend`` choice does: after checking some preconditions, it archives the current user contract and creates a new one with the extra friend added to the list. Here is a quick explanation of the code: 
 
-There is some boilerplate to set up the choice (full details in the :doc:`DAML reference </daml/reference/choices>`):
+    - The choice starts with the ``nonconsuming choice`` keyword followed by the choice name, which is in this case ``AddFriend``.
+    - The return type of a choice is defined next. In this case it is ``ContractId User``.
+    - After that we pass arguments for the choice with ``with`` keyword and for ``AddFriend`` this is  ``friend: Party``.
+    - The keyword ``controller`` defines the ``Party`` that is allowed to execute the choice. In this case it would be the logged in user with the defined ``username``.  
+    - The ``do`` keyword marks the start of the choice's body where its functionality will be written.
+    - After passing some checks current contract is archived with ``archive self`` 
```

Comment:
I wonder if it would make sense to point out that it doesn’t matter if the checks come before or after archiving `self` or maybe move the archive to the first line. Currently users might get the mistaken impression that doing the checks afterwards would be dangerous since the contract might get archived even if the checks fail. Since the choice is transactional this is not the case. Probably not super important for the getting started guide.

Reply:
Will write your comment down and leave it as is (just because it's the GSG). The explanation now follows what's in the code.

Reply:
I think it's good practice to have the checks at the top of the choice so they are easy to find. I'm actually in favor of archiving `self` in the line before creating the new contract. This makes it more obvious that we're "updating" the contract.

This line should also end with a period.